cassandra/src/test/java/org/hibernate/ogm/datastore/cassandra/test/query/nativequery/CassandraEntityManagerNativeQueryTest.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.cassandra.test.query.nativequery;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.hibernate.ogm.backendtck.jpa.Poem;
+import org.hibernate.ogm.utils.PackagingRule;
+import org.hibernate.ogm.utils.TestForIssue;
+import org.hibernate.ogm.utils.jpa.JpaTestCase;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * Test the execution of native queries on Cassandra using the {@link EntityManager}
+ *
+ * @author Davide D'Alto &lt;davide@hibernate.org&gt;
+ * @author Jonathan Halliday
+ */
+public class CassandraEntityManagerNativeQueryTest extends JpaTestCase {
+
+	@Rule
+	public PackagingRule packaging = new PackagingRule( "persistencexml/ogm.xml", Poem.class );
+
+	private final OscarWildePoem portia = new OscarWildePoem( 1L, "Portia", "Oscar Wilde" );
+	private final OscarWildePoem athanasia = new OscarWildePoem( 2L, "Athanasia", "Oscar Wilde", (byte) 2 );
+	private final Critic critic = new Critic( new CriticId( "de", "764" ), "Roger" );
+
+	private EntityManager em;
+
+	@Before
+	public void init() throws Exception {
+		// prepare test data
+		em = createEntityManager();
+		begin();
+		em = persist( portia, athanasia, critic );
+		commit();
+		em.close();
+
+		em = createEntityManager();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		begin();
+		delete( portia, athanasia, critic );
+		commit();
+		close( em );
+	}
+
+	@Test
+	public void testSingleResultQuery() throws Exception {
+		begin();
+
+		String nativeQuery = "SELECT * FROM \"WILDE_POEM\" WHERE name='Portia'";
```

Comment:
IIRC CASSANDRA-8393 was the straw that broke the camel's back, but in general c* automatically lower cases anything not quoted, whilst the driver's internals don't. Because we're having to play funny games with concatenating identifiers, it's also possible that something which starts out looking innocuous then winds up containing metacharacters and thus requiring quoting. It's basically a lot simpler to quote everything for safety than to derive a foolproof set of rules for when quoting is required.  Maybe open a JIRA for it and we'll revisit if it gets enough votes?